src/expr/src/scalar/func.rs

```diff
@@ -2071,6 +2071,18 @@ where
     }
 }
 
+fn date_trunc_interval<'a>(a: Datum, mut int: Interval) -> Result<Datum<'a>, EvalError> {
+    let units = a.unwrap_str();
+    let dtf = units
+        .parse()
+        .map_err(|_| EvalError::UnknownUnits(units.to_owned()))?;
+
+    // truncate_low_fields should never return an error in our case
+    int.truncate_low_fields(dtf, Some(0))
```

Comment:
if something should never return an error/is an unrecoverable logic error, you can call `.expect("<why I should never error>")`. `EvalError::Internal` is slightly different in that it's meant to bubble up an internal error--there are a few places in the code that I think should be `expect` using this, so seems like it could use some documentation or guidance.

Reply:
i also did that